addons/portal/static/src/js/portal_name_and_signature.js

```diff
@@ -0,0 +1,582 @@
+        this.$signatureGroup.toggleClass('border-danger', isSignatureEmpty);
+        return name && !isSignatureEmpty;
+    },
+
+    //----------------------------------------------------------------------
+    // Private
+    //----------------------------------------------------------------------
+
+    /**
+     * Draws the current name with the current font in the signature field.
+     *
+     * @private
+     */
+    _drawCurrentName: function () {
+        var font = this.fonts[this.currentFont];
+        var text = this._getCleanedName();
+        var canvas = this.$signatureField.find('canvas')[0];
+        var img = this._getSVGText(font, text, canvas.width, canvas.height);
+        return this._printImage(img);
+    },
+    /**
+     * Returns the given name after cleaning it by removing characters that
+     * are not supposed to be used in a signature. If @see signatureType is set
+     * to 'initial', returns the first letter of each word, separated by dots.
+     *
+     * @private
+     * @returns {string} cleaned name
+     */
+    _getCleanedName: function () {
+        var text = this.getName().replace(/[^[\w\u00E0-\u00FC]-'" ]/g, '');
+        if (this.signatureType === 'initial') {
+            return (text.split(' ').map(function (w) {
+                return w[0];
+            }).join('.') + '.');
+        }
+        return text;
+    },
+    /**
+     * Gets an SVG matching the given parameters, output compatible with the
+     * src attribute of <img/>.
+     *
+     * @private
+     * @param {string} font: base64 encoded font to use
+     * @param {string} text: the name to draw
+     * @param {number} width: the width of the resulting image in px
+     * @param {number} width: the height of the resulting image in px
+     * @returns {string} image = mimetype + image data
+     */
+    _getSVGText: function (font, text, width, height) {
+        var $svg = $(core.qweb.render('portal.sign_svg_text', {
+            width: width,
+            height: height,
+            font: font,
+            text: text,
+            type: this.signatureType,
+        }));
+        $svg.attr({
+            'xmlns': "http://www.w3.org/2000/svg",
+            'xmlns:xlink': "http://www.w3.org/1999/xlink",
+        });
+
+        return "data:image/svg+xml," + encodeURI($svg[0].outerHTML);
+    },
+    /**
+     * Displays the given image in the signature field.
+     * If needed, resizes the image to fit the existing area.
+     *
+     * @private
+     * @param {string} imgSrc - data of the image to display
+     */
+    _printImage: function (imgSrc) {
+        var self = this;
+
+        var image = new Image;
+        image.onload = function () {
+            // don't slow down the UI if the drawing is slow, and prevent
+            // drawing twice when calling this method in rapid succession
+            clearTimeout(self.drawTimeout);
+            self.drawTimeout = setTimeout(function () {
```

Comment:
Note: was this like that before ? Feels strange to me (why would this be called in rapid succession?). Also if this is needed, it would have been even better to put the final function aside so that the behavior can be extended (here if you want to extend `_printImage`, you can't). Anyways, this can stay like that, just a note.

Reply:
I think it wasn't like that before but without this I had an issue when typing my name too fast.

The problem was especially bad when I used to show all the styles too at the same time before fixing the absolute right panel.